### ISSUE TITLE
Add support for proxy setup

### ISSUE THREAD

**Issue by tyler-ball** [chef/chef-analyze#69]
### Describe the Enhancement:
As a `chef-analyze` user,
I expect the command to respect my workstation's proxy configuration,
so I can run the command in a locked down environment that requires proxies.

### Describe the Need:
Particularly for our enterprise customers we need to add proxy support. In the Ruby ecosystem we developed a gem that knows how to ready proxy settings from chef config and environment variables, merges them, and exports the merged proxy config as an environment variable in the scope of the running command. We did this because our research showed that environment variables are the most widely understood / respected proxy setup for our tools and dependencies we call out to.

A minimum implementation would be for `chef-analyze` to respect proxy environment variables and system level proxy settings on MacOS/Windows.

It should also have automated testing to ensure that we correctly support proxies, but that greatly increases the complexity of this card.

### Current Alternative
We need to look into whether our current Golang libraries already respect reading proxy environment variables. If they do we can decide whether to attempt to parse the Ruby config (EG, `~/.chef/config.rb`) to do the proxy settings merging that currently occurs in our Ruby products.

**Comment by afiune** [comment]
Yes, we absolutely need this feature. I will add a Critical Medium just to keep it up in the priority list.
